strawberryfields/api/devicespec.py

```diff
@@ -87,6 +97,24 @@ def gate_parameters(self):
 
         return gate_parameters
 
+    def validate_parameters(self, **parameters):
+        """Validate gate parameters against the device spec.
+
+        Gate parameters should be passed as keyword arguments, with names
+        correspond to those present in the Blackbird circuit layout.
+        """
+        # check that all provided parameters are valid
+        for p, v in parameters.items():
+            if p in self.gate_parameters and v not in self.gate_parameters[p]:
+                # parameter is present in the device specifications
+                # but the user has provided a disallowed value
+                raise ValueError(
+                    f"{p} has invalid value {v}. Only {self.gate_parameters[p]} allowed."
+                )
+
+            if p not in self.gate_parameters:
+                raise ValueError(f"Parameter {p} not a valid parameter for this device")
```

Comment:
(Minor suggestion): wondering if it was nice to log something if the parameters are valid :thinking: 
`self.log.info("The validation of parameters was completed.")`

Not entirely convinced (doesn't seem super crucial), more of an idea. If decided on, a logger object can be set up for `DeviceSpec` instances similar to how it is managed in `connection.py`.